src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -538,6 +574,66 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
         err
     }
 
+    fn find_name_and_ty_of_values(
+        &self,
+    ) -> Option<(String, Span, String, Span)> {
+        let mir_node_id = self.infcx.tcx.hir.as_local_node_id(self.mir_def_id)?;
+        let fn_decl = self.infcx.tcx.hir.fn_decl(mir_node_id)?;
+
+        // If there is one argument and this error is being reported, that means
+        // that the lifetime of the borrow could not be made to match the single
+        // argument's lifetime. We can highlight it.
+        //
+        // If there is more than one argument and this error is being reported, that
+        // means there must be a self parameter - as otherwise there would be an error
+        // from lifetime elision and not this. So we highlight the self parameter.
+        let arg = fn_decl.inputs
+            .first()
+            .and_then(|ty| {
+                if let hir::TyKind::Rptr(
+                    hir::Lifetime { name, span, ..  }, ..
+                ) = ty.node {
+                    Some((name, span))
+                } else {
+                    None
+                }
+            });
+
+        let ret = if let hir::FunctionRetTy::Return(ret_ty) = &fn_decl.output {
+            if let hir::Ty {
+                node: hir::TyKind::Rptr(hir::Lifetime { name, span, ..  }, ..),
+                ..
+            } = ret_ty.clone().into_inner() {
+                Some((name, span))
+            } else {
+                None
+            }
+        } else {
+            None
+        };
+
+        let (arg_name, arg_span) = arg?;
+        let (return_name, return_span) = ret?;
+
+        let lifetimes_match = arg_name == return_name;
+
+        let arg_name = if arg_name.is_elided() {
+            "'0".to_string()
+        } else {
+            format!("{}", arg_name.ident())
+        };
```

Comment:
Would it make sense to re-use that enum you created before?

Reply:
This code doesn't exist in recent commits.